c10/core/impl/TorchDispatchModeTLS.cpp

```diff
@@ -50,7 +53,15 @@ const TorchDispatchModeTLS& TorchDispatchModeTLS::get_state() {
 }
 
 void TorchDispatchModeTLS::set_state(const TorchDispatchModeTLS& state) {
+  for (const std::shared_ptr<c10::ModePyObjTrampoline>& state :
```

Comment:
@eellison JFYI, contrary to what I previously said, this change ended up having to be thread safe. So once fake tensor mode is thread safe, the checks on the mode stack should also be thread safe and it should be good to go